streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractStateManager.java

```diff
@@ -45,8 +45,7 @@
     OffsetCheckpoint checkpoint;
 
     final Map<TopicPartition, Long> checkpointableOffsets = new HashMap<>();
-    final Map<String, StateStore> stores = new LinkedHashMap<>();
-    final Map<String, StateStore> globalStores = new LinkedHashMap<>();
+    final FixedOrderMap<String, Optional<StateStore>> globalStores = new FixedOrderMap<>();
```

Comment:
Could we move `globalStores` to GlobalStateManagerImpl as well?

Reply:
No, because the ProcessorStateManager offers an interface to look up global stores, as well (for some reason, maybe joins?)

Reply:
Ack.